packages/react/src/components/Forms/DatePicker/Calendar.tsx

```diff
@@ -214,6 +183,28 @@ export const Calendar = ({
     ...nextButtonOtherProps
   } = nextButtonProps;
 
+  const getToggleButtonProps = (
+    key: string,
+    items: Array<Option>,
+    downshift: UseSelectReturnValue<Option>
+  ) => ({
+    ...{
+      ...downshift.getToggleButtonProps(),
+      onClick: () => {
+        const selectedItem = items.find(
+          (item) => item.value === state.focusedDate[key as keyof CalendarDate]
+        );
+        if (selectedItem) {
+          downshift.selectItem(selectedItem);
+        }
+        downshift.toggleMenu();
+      },
+      "aria-label": t(
+        `components.forms.date_picker.${key}_select_button_aria_label`
+      ),
+    },
+  });
```

Comment:
You could remove one level.
```suggestion
  ) => ({
    ...downshift.getToggleButtonProps(),
    onClick: () => {
      const selectedItem = items.find(
        (item) => item.value === state.focusedDate[key as keyof CalendarDate]
      );
      if (selectedItem) {
        downshift.selectItem(selectedItem);
      }
      downshift.toggleMenu();
    },
    "aria-label": t(
      `components.forms.date_picker.${key}_select_button_aria_label`
    ),
  });
```